File: BinarySearch.cpp
#include <iostream>
using namespace std;

int binarySearch(int a[], int n, int val)
{
    int start = 0, end = n - 1;
    while (start <= end)
    {
        int mid = (start + end) / 2;

        if (a[mid] == val)
        {
            return mid;
        }

        else if (val < a[mid])
        {
            end = mid - 1;
        }

        else
        {
            start = mid + 1;
        }
    }

    return -1;
}

int main()
{
    int n;
    cout << "ENter the size of array : ";
    cin >> n;
    int a[100];

    cout << "Enter the elements of the array : ";

    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }

    int x;
    cout << "Enter the target element : ";
    cin >> x;

    cout << "Target Element is found at location : "<< binarySearch(a, n, x);

    return 0;
}

File: Binary Search.py
def binary_search(l,low,high,val):
    if high<low:
        return None
    else:
        mid=low+((high-low)//2)
        
        if l[mid]>val:
            return binary_search(l,low,mid-1,val)
        elif l[mid]<val:
            return binary_search(l,mid+1,high,val)
        else:
            return mid
l=[12,23,34,43,56,67,78,89,98]
print(binary_search(l,0,8,56))
